src/cha.c

```diff
@@ -409,7 +433,17 @@ int cha_start_stream(struct cha* cha) {
 	if (ret == -1)
 		return ret;
 
-	ret = cha_write_reg_by_name(cha, CSTREAM_CFG, 1);
+	ret = cha_cast_reg_by_name(cha, CSTREAM_CFG, 1);
+	if (ret == -1)
+		return ret;
+
+	return 0;
+}
+
+int cha_halt_stream(struct cha* cha) {
+	int ret = 0;
+
+	ret = cha_cast_reg_by_name(cha, SDRAM_HOST_READ_GO, 0);
```

Comment:
If you want to see `HF0_LAST` you should set `CSTREAM_CFG` to 0 here and leave `SDRAM_HOST_READ_GO` intact. Note that the `HF0_LAST` packet in current gateware will be stalled in the out fifo (unless you are lucky to reach the SDRAM burst size with the last packet) - it is fixed with the new gateware. There is no real reason to set `SDRAM_HOST_READ_GO` to 0 other than stressing out gateware.

Reply:
Yes, I did exactly like that and stream just stopped without any `HF0_LAST`.

Reply:
But you should eventually see `HF0_LAST` after setting `CSTREAM_CFG` back to 1.

Reply:
I would like to have `ov_capture_shutdown()` function which send `CSTREAM_CFG=0` and then `ov_capture_dispatch()` terminates on `HF0_LAST`:
 * `ov_capture_breakloop()` for people who wants immediate interrupt
 * `ov_capture_shutdown()` for people who wants to collect all data

However, with the current version of the firmware `ov_capture_shutdown()` is impossible because it only leads to the hung due to missed  `HF0_LAST`.